zinc/src/test/scala/sbt/inc/BinaryDepSpec.scala

```diff
@@ -0,0 +1,26 @@
+package sbt.inc
+
+import sbt.internal.inc.Analysis
+import sbt.io.IO
+
+class BinaryDepSpec extends BaseCompilerSpec{
+  it should "not depend on non-existing objects" in {
```

Comment:
Is this expectation true? What if you changed the macro? Shouldn't we invalidate the macro consumer?

Reply:
We do not change macro here we compile each project once. 

We depend on NoopMacro but as a external dependency and we should not depended at the same time on it as library dependency (so we don't want recompile every usage of NoopMacro when we change anything in jar where it comes from)